lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -608,20 +667,51 @@
         return toReturn;
     }());
 
+    // mappedDefaultLocale will get the default locale and update any deprecated
+    // collation/sort order values it may use
+    let __mappedDefaultLocale = undefined;
     var mappedDefaultLocale = function () {
-        var parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
-        var locale = parts[1];
-        var collation = parts[2];
-        var availableBcpTags = reverseLocaleAcceptingCollationValues[locale];
+        if (__mappedDefaultLocale) {
+            return __mappedDefaultLocale;
+        }
 
-        if (collation === undefined || availableBcpTags === undefined) { return locale; }
+        let locale = undefined;
+        let collation = undefined;
+        if (isPlatformUsingICU) {
+            // ICU's getDefaultLocale() will return a valid BCP-47/RFC 5646 langtag
+            locale = GetDefaultLocale();
+            const match = platform.builtInRegexMatch(locale, /-u(-[^\-][^\-]?-[^\-]+)*-co-([^\-]+).*/);
+            if (match) {
+                collation = match[2];
+                locale = callInstanceFunc(StringInstanceReplace, locale, `-co-${match[2]}`, "");
+            }
+        } else {
+            // Windows' getDefaultLocale() will return a weird RFC4646 langtag
+            const parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
+            locale = parts[1];
+            collation = parts[2];
+        }
+
+        if (collation === undefined) {
+            __mappedDefaultLocale = locale;
+            return __mappedDefaultLocale;
+        }
 
-        var bcpTag = availableBcpTags[collation];
-        if (bcpTag !== undefined) {
-            return locale + "-u-co-" + bcpTag;
+        const collationMapForLocale = reverseLocaleAcceptingCollationValues[locale];
```

Comment:
>reverseLocaleAcceptingCollationValues [](start = 38, length = 37)

I think the whole point of these mappings is to translate RFC 4646 (or associated standard/database/list) collation values into RFC 5646 (or associated) collation values. See the Intl explainer for the RFC 5646-associated list.
IOW this shouldn't be necessary for ICU because the langtag should already be RFC 5646.
However, since this value is in theory reported by the system, it might still be a problem. I'd like to think that ICU would convert or ignore anything it gets from Windows that isn't valid according to ICU.

Reply:
Doing a lookup in a dictionary once during the lifetime of the program is worth not needing to worry about this, I think. We may want to re-evaluate that decision if we make these dictionaries into a native-calculated, hardcoded thing (then, we could just not have this dictionary at all if ICU is enabled).